chipsec/chipset.py

```diff
@@ -606,24 +283,92 @@ def print_pch(self):
+    def init_xml_configuration(self):
+        self.pch_dictionary = dict()
+        self.chipset_dictionary = dict()
+        self.device_dictionary = dict()
+        self.chipset_codes = {}
+        self.pch_codes = {}
+        self.device_code = []
+        self.load_list = []
+
+        # find VID
+        _cfg_path = os.path.join( chipsec.file.get_main_dir(), 'chipsec', 'cfg' )
+        VID = [f for f in os.listdir(_cfg_path) if os.path.isdir(os.path.join(_cfg_path, f)) and not f[:2] == '__' ]
+        # create dictionaries
+        for vid in VID:
+            self.chipset_dictionary[int(vid,16)] = collections.defaultdict(list)
+            self.pch_dictionary[int(vid,16)] = collections.defaultdict(list)
+            self.device_dictionary[int(vid,16)] = collections.defaultdict(list)
+            for fxml in os.listdir(os.path.join(_cfg_path,vid)):
+                if logger().DEBUG: logger().log( "[*] looking for platform config in '{}'..".format(fxml) )
+                tree = ET.parse( os.path.join(_cfg_path,vid,fxml) )
+                root = tree.getroot()
+                for _cfg in root.iter('configuration'):
+                    if 'platform' not in _cfg.attrib:
+                        if logger().DEBUG: logger().log( "[*] found common platform config '{}'..".format(fxml) )
+                        self.load_list.append(fxml)
+                        continue
+                    elif _cfg.attrib['platform'].lower().startswith('pch'):
+                        if logger().DEBUG: logger().log( "[*] found PCH config at '{}'..".format(fxml) )
+                        self.pch_codes[_cfg.attrib['platform'].lower()] = {}
+                        self.pch_codes[_cfg.attrib['platform'].lower()]['vid'] = int(vid,16)
+                        mdict = self.pch_dictionary[int(vid,16)]
+                        cdict = self.pch_codes[_cfg.attrib['platform'].lower()]
+                    elif _cfg.attrib['platform'].lower():
+                        if logger().DEBUG: logger().log("[*] found platform config from '{}'..".format(fxml))
+                        self.chipset_codes[_cfg.attrib['platform'].lower()] = {}
+                        self.chipset_codes[_cfg.attrib['platform'].lower()]['vid'] = int(vid,16)
+                        mdict = self.chipset_dictionary[int(vid,16)]
+                        cdict = self.chipset_codes[_cfg.attrib['platform'].lower()]
+                    else: continue
+                    if logger().DEBUG: logger().log( "[*] Populating configuration dictionary.." )
+                    for _info in _cfg.iter('info'):
+                        if 'family' in _info.attrib:
+                            if _info.attrib['family'].lower() == "core":
+                                CHIPSET_FAMILY_CORE.append(_cfg.attrib['platform'].lower())
+                            if _info.attrib['family'].lower() == "atom":
+                                CHIPSET_FAMILY_XEON.append(_cfg.attrib['platform'].lower())
```

Comment:
you have swapped xeon and atom...

Reply:
Fixed in #871.  I can fix it here too if necessary